### PR TITLE
Add more stuff about debugging

### PR DESCRIPTION
**Pull request by liach**

### CODE REVIEW

**Comment by ST-DDT** [comment]
Addresses: #356

**Comment by ST-DDT** [comment]
@liach Any updates on this?

**Comment by liach** [comment]
Unfortunately nope. I am currently in china and cannot even log into discord (it asks me to check email while i cannot log on to my email service in china)

**Comment by Inscrutable** [comment]
This is gathering dust @liach - will you be able to complete it soon?

**Comment by liach** [comment]
Unfortunately no. I cannot find big chunks of time to devote on stuff.

**Comment by ST-DDT** [comment]
@liach Is the current part complete enough to be merged - so that we have at least this instead of nothing?
We can always PR the remaining stuff later.

**Comment by liach** [comment]
i'd get a developer to confirm the validity of this portion.

**Review by ST-DDT — APPROVED** [review]

File: source/contributing/implementation/debugging.rst [hunk 1]
```diff
@@ -4,6 +4,39 @@ Debugging Sponge Within the IDE
...
+Unfortunately, SpongeCommon exists as an abstract layer to reduce duplicate code between SpongeForge and SpongeVanilla;
+thus, it is not for production environments, and you cannot run SpongeCommon directly. However, you can run SpongeForge
+or SpongeVanilla with the changes you've applied in SpongeCommon. Below is a description on testing SpongeCommon changes
+on SpongeForge.
+
+Assume you have a branch ``fix/foo``, based on ``bleeding`` branch of ``SpongePowered/SpongeCommon``, on your
+SpongeCommon fork. You want to test it against the corresponding SpongeForge (or the ``bleeding`` branch of it). The
+procedure is as follows:
+
+1. Clone SpongeForge repository from GitHub: Find a clean workspace, run
```
Comment (Inscrutable) [inline source/contributing/implementation/debugging.rst:25]: I wonder if ReST will interpret this as three lists and start each one at `1.`. We shall see...

Comment (liach) [reply source/contributing/implementation/debugging.rst:25]: I recall @Spongy builds prs automatically. Doesn't it?

Comment (liach) [reply source/contributing/implementation/debugging.rst:25]: Hmm. appears spongy doesn't like me (no branch `661`)

**Comment by phit** [comment]
that seems like a rather complicated way of doing this all, why would you have a SC branch without already having a SF or SV workspace?

what I think most here do, setup SF or SV as usual, then cd into the SpongeCommon folder and create a new branch.. really I don't see the point of anything you added here

this page if anything should simply have the sf/sv setup instructions that are also in the main repo readmes followed by what is already covered in the plugin debugging page https://docs.spongepowered.org/stable/en/plugin/debugging.html#running-the-configuration

**Comment by Spongy** [comment]
A preview for this pull request is available at https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/2a7e549/.

Here are some links to the pages that were modified:

- modified: https://cdn.rawgit.com/Spongy/SpongeDocs-PRs/2a7e549/contributing/implementation/debugging.html

_Since the preview frequently changes, please link to [this comment](https://github.com/SpongePowered/SpongeDocs/pull/661#issuecomment-439694772), not to the direct url to the preview._

**Comment by liach** [comment]
Closing in favor of #784 as suggested by inscrutable

**Pull request closed without merging**